src/src/PluginStructs/P116_data_struct.cpp

```diff
@@ -105,6 +105,26 @@ P116_data_struct::P116_data_struct(ST77xx_type_e       device,
   _commandTriggerCmd += F("cmd");
 }
 
+/****************************************************************************
+ * Destructor
+ ***************************************************************************/
+P116_data_struct::~P116_data_struct() {
+  if (nullptr != gfxHelper) {
+    delete gfxHelper;
+    gfxHelper = nullptr;
+  }
+
+  if (nullptr != st77xx) {
+    delete st77xx;
+    st77xx = nullptr;
+  }
+
+  // No need to delete these, as the object has been deleted via st77xx already, see plugin_init
+  st7735 = nullptr;
```

Comment:
That's a bold statement.
A quick glance only shows it is deleted in the call of `PLUGIN_EXIT`
But is there a guarantee that's being called?
Why not check for the pointer value and delete it anyway?

Or maybe we should start using some kind of unique pointer (or one of the many similar 'smarter' pointer types) in our data types.

Reply:
The st7735/89/96 pointers are only used to create and initialize the object, then it is assigned to st77xx, so when `delete st77xx` is executed, the temporary pointers are pointing to an already released object. Would possibly crash when deleted, but I think to remember that delete is very careful in trying to not cause crashes. Maybe I should set it to `nullptr` once I no longer need it.

Reply:
Yep, I noticed you do assign the pointer to the `st77xx` pointer, so that one shouldn't be deleted.
But the other ones should be deleted and I only saw that in the PLUGIN_EXIT call, or did I miss another ?

Reply:
Ah, I may have mixed something up here, I'll re-do my homework then